backends/dpdk/dpdkArch.cpp

```diff
@@ -467,6 +467,127 @@ bool CollectMetadataHeaderInfo::preorder(const IR::Type_Struct *s) {
     return true;
 }
 
+const IR::Node *AlignHdrMetaField::preorder(IR::Type_StructLike *st) {
+    auto fields = new IR::IndexedVector<IR::StructField>;
+    if (st->is<IR::Type_Header>()) {
+        auto h = st->to<IR::Type_Header>();
+        unsigned size_sum_so_far = 0;
+        for (auto field : st->fields) {
+            if (auto t = (*field).type->to<IR::Type_Bits>()) {
```

Comment:
Please add a check if the type->is<IR::Type_Bits>, If there is a structure within another structure, this might result in seg fault

Reply:
Addressed